### PR TITLE
ghp-import: 0.5.5 -> 1.1.0

### PR DESCRIPTION
**Pull request by veehaitch**
###### Motivation for this change

- Updated package: 0.5.5 -> 1.1.0
- Converted to a Python package to allow using it as a library as well
- URL and license changed
- Added myself as maintainer (reluctantly though)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120794` at 38c3e6a9 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>ghp-import (python38Packages.ghp-import)</li>
    <li>python39Packages.ghp-import</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/ghp-import/default.nix:21:0:
```
   |
21 |     description = "Copy your docs directly to the gh-pages branch";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/ghp-import/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+, lib
+}:
+
+buildPythonPackage rec {
+  pname = "ghp-import";
+  version = "1.1.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "wiqc4Qw3dT4miNFk12WnANrkuNefptsKLDEyuniBiU8=";
```
Comment (drewrisinger) [inline pkgs/development/python-modules/ghp-import/default.nix:13]: Recent best practice I've seen has attached the SRI hash-type indicator, see e.g. r-ryantm updates. Something like ``hash = "sha256-...";

File: pkgs/development/python-modules/ghp-import/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{ buildPythonPackage
+, dateutil
+, fetchPypi
+, lib
```
Comment (drewrisinger) [inline pkgs/development/python-modules/ghp-import/default.nix:4]:
```suggestion
{ lib
, buildPythonPackage
, fetchPypi
, dateutil
```
Conventionally, lib & buildPythonPackage & fetchpypi go first.

File: pkgs/development/python-modules/ghp-import/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    sha256 = "wiqc4Qw3dT4miNFk12WnANrkuNefptsKLDEyuniBiU8=";
+  };
+
+  pythonImportsCheck = [ "ghp_import" ];
+
+  propagatedBuildInputs = [ dateutil ];
+
+  meta = with lib; {
+    description = "Copy your docs directly to the gh-pages branch";
+    homepage = "https://github.com/c-w/ghp-import";
```
Comment (drewrisinger) [inline pkgs/development/python-modules/ghp-import/default.nix:22]: Homepage changed b/c username appears to have changed. The old URL redirects here 🆗

File: pkgs/development/python-modules/ghp-import/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    license = licenses.asl20;
```
Comment (drewrisinger) [inline pkgs/development/python-modules/ghp-import/default.nix:23]: License matches current license on github site.

**Review by drewrisinger — APPROVED** [review]
* [x] Diff mostly LGTM
* [x] Commits LGTM
* [x] Builds via ``nix-review`` on x86-64-linux:

```
2 packages built:
ghp-import python39Packages.ghp-import
```

File: pkgs/development/python-modules/ghp-import/default.nix (hunk 2 above)
Comment (veehaitch) [reply pkgs/development/python-modules/ghp-import/default.nix]: Done.

File: pkgs/development/python-modules/ghp-import/default.nix (hunk 1 above)
Comment (veehaitch) [reply pkgs/development/python-modules/ghp-import/default.nix]: Done.

**Comment by veehaitch** [comment]
Feel free to squash on merge.

**Pull request merged**